keps/sig-node/3066-subsecond-probes/README.md

```diff
@@ -0,0 +1,866 @@
+know that this has succeeded?
+-->
+An ability to specify timeouts that are less than one second.
+Add additional tests cases to the timeout test cases.
+
+? link to existing test cases ?
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+V2 API for existing objects.
+Converting fields from int32 to resource.Quantity.
+Subsecond resolution less than one millisecond.
+
+## Proposal - Plan-A
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+Add a new int32 field to existing Probe struct for timeoutSeconds that exists of timeoutMilliseconds.
+
+int32 data type is used for consistency with existing fields.
+
+Using the fully available range of an int32 with a maximum value of 2147483647 represents a value of ~24.8 days.
+
+If the Milliseconds variant of a field is set,
+*use it in preference* to the existing Seconds field,
+and **completely ignore** the value of the existing field.
+This behavior makes it opt-in on setting a non-zero Milliseconds field.
+
+## Proposal - Plan-B
+
+Add a new field: ReadSecondsAs string (“seconds” || “milliseconds”) where default "" translates to "seconds"
+
+If using “milliseconds” the minimums would need to be no less than 100millseconds
+and defaults will be set to existing values in seconds.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Knative
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Changing defaults is a strict no-go.
+
+
+
+Accidentally setting a timeout too low could DOS kubelet if many are used.
+Mitigate by preventing timeout values too small.
+Could be configurable,
+10-100milliseconds is a first guess.
+
+UX reviewed by existing users of Probe struct.
+? add details on who that is?
+
+#### Overriding an existing field
+
+How does the change to overriding a field effect the users of the existing field.
```

Comment:
```suggestion
How does the change to overriding a field effect the users of the existing field.

Currently, the time fields in the Probe struct only use values in seconds. Since this KEP proposes allowing a change of the units on those fields, that could be considered an override. 

However, this change would be done on a probe by probe basis and would be opt-in. Meaning that the default behavior would not change unless the probe specifically added the `ReadSecondsAs` field to their spec. Absent that specific opt-in by using the new field, existing users should not be impacted.
```